gradoop-examples/src/main/java/org/gradoop/examples/AbstractRunner.java

```diff
@@ -189,4 +197,52 @@ protected static void convertDotToPNG(String dotFile, String pngFile) throws IOE
     pb.redirectOutput(ProcessBuilder.Redirect.appendTo(output));
     pb.start();
   }
+
+  /**
+   * Returns an EPGM DataSource for a given directory and format.
+   *
+   * @param directory input path
+   * @param format    format in which the data is stored (csv, indexed, json)
+   * @return DataSource for EPGM Data
+   * @throws IOException
+   */
+  private static DataSource getDataSource(String directory, String format) throws IOException {
+    directory = appendSeparator(directory);
+
+    GradoopFlinkConfig config = GradoopFlinkConfig.createConfig(getExecutionEnvironment());
+    format = format.toLowerCase();
+
+    if (format.equals("json")) {
```

Comment:
Maybe we can store a static `Map<String, Class<? extends DataSource>>` to map a format to its corresponding class. If the parameter `format` is not in the map throw `IllegalArgumentException`. Same for getDataSink. Also with `containsKey()` the validity of a given format can be checked.

Reply:
In order to call the constructor with arguments from the class object, we would need to use reflection. Calling the constructor from the class directly with `class.newInstance()` does not allow any arguments.